source/agora/node/Ledger.d

```diff
@@ -432,6 +424,25 @@ public class Ledger
         return to!ulong(tx_amount.toString());
     }
 
+    /***************************************************************************
+
+        Update the ledger state from a block which was read from storage
+
+        Params:
+            block = block to update the state from
+
+    ***************************************************************************/
+
+    protected void updateStateFromStorageBlock (in Block block) @safe
```

Comment:
I would think that encapsulating the whole `if` would be better, unless you have other plans ?
E.g.
```D
if (this.utxo_set.length == 0)
    this.updateStateFromStorageBlock();
```

Also... `this.replayBlockchain()` or something a bit more succinct ? :D

Reply:
I need to have it called per block but I can rename it.